File: 260-single-number-III/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> Vec<i32> {
        
        let mut count_map: HashMap<i32, i32> = HashMap::new();

        for &num in &nums {
            *count_map.entry(num).or_insert(0) += 1;
        }

        let mut res: Vec<i32> = Vec::new();
        for (&k, &v) in count_map.iter() {
            if v == 1 {
                res.push(k);
            }
        }

        res
    }
}

File: 260-single-number-iii/main.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:

        d = defaultdict(int)
        for num in nums:
            d[num] = d.get(num, 0) + 1

        res = [k for k, v in d.items() if v == 1]
    
        return res
